sdk/core/azure-core/src/http/request.cpp

```diff
@@ -10,6 +10,18 @@
 
 using namespace Azure::Core::Http;
 
+namespace {
```

Comment:
Why anonymous namespace instead of `Details`?

Reply:
The function is local to the compilation unit (this cpp file).  We don't use details in those cases.

Reply:
In that case, why not just use a private static method that isn't exposed via a header? Wouldn't that be local to this file anyway, without having it in a separate anonymous namespace?

Reply:
We agreed to use unnamed namespaces instead of that.
There are some benefits of doing this on C++ vs the static methods...
The one I consider most important is preventing same-name static methods declared in different cpp files.
See this:
https://stackoverflow.com/questions/4422507/superiority-of-unnamed-namespace-over-static

Reply:
whenever possible, we prefer unnamed namespaces.  
There might be some static functions in cpp files from before we agreed to use the anonymous namespace. Those should be eventually updated.

Reply:
In that case, @CaseyCarter, we should update the guidelines to mention that: https://azure.github.io/azure-sdk/cpp_design.html#namespaces

Reply:
https://github.com/Azure/azure-sdk/issues/1937